src/hotspot/share/c1/c1_Instruction.cpp

```diff
@@ -835,7 +871,17 @@ bool BlockBegin::try_merge(ValueStack* new_state) {
           // In really rare cases we will bail out in LIRGenerator::move_to_phi.
           existing_phi->make_illegal();
           existing_state->invalidate_local(index);
-          TRACE_PHI(tty->print_cr("invalidating local %d because of type mismatch", index));
+          TRACE_PHI(tty->print_cr("invalidating local %d because of type mismatch%s",
+                                  index, new_value == NULL ? " (new_value is NULL)" : ""));
+          // Check if illegal phi gets used.
+          SearchUsageClosure search(existing_phi);
```

Comment:
should we add something like "FIXME" comment to indicate this is a workaround and we need further investigation to solve it?